documents/dev_designs/account_management.md

```diff
@@ -0,0 +1,87 @@
+# Integrating with Steam and other game store accounts
+
+# Requirements
+
+## Seamless Steam Accounts
+
+- Users launching BAR on Steam must be able to login for online play without
+  the need for registration.
+  - This user experience is unlike the current flow which requires registration
+  prior to online play.
+- Users must be able to manage their account on server4.beyondallreason.info.
+  - If the user has never registered automatically by launching BAR on Steam,
+  server4.beyondallreason.info must provide a registration flow with Steam as
+  the ID provider.
+  - If the user has registered via Steam, a 'login with steam' button must be
+  provided on server4.beyondallreason.info.
+
+## Upgrade Path For Current Players
+
+- Players with pre-existing BAR accounts (henceforce called 'legacy accounts')
+  must be able to migrate seamlessly onto Steam. To comply with the
+  [no smurfs rule](https://www.beyondallreason.info/code-of-conduct#5-unfair-advantages)
+  note the following:
+  - Players may create a new account linked to Steam and link the newly created
+    Steam account with their legacy account.
+      - Legacy account owners may unknowingly start BAR on Steam and play with
+      the new Steam account prior to linking their legacy accounts - resulting
+      in unintentional 'smurfs'. We must provide clear policy guidance, e.g. a
+      possible grace period for 'accidently' playing on these smurfs, send
+      notifications for these 'smurfs' to link their accounts.
+  - Alternately, players must be able to log into their legacy accounts via
+    server4.beyondallreason.info and link to a Steam account.
+
+## (Stretch) Support For Additional Game Stores
+
+- We may want to also have the same Steam experience with other game stores
+  (e.g. Epic Games).
+  - The design should allow any account to be linked to any number of additional game stores.
+
+## (Stretch) Support For Unlinking
+
+- We may provide the ability for players to request accounts to be unlinked.
+
+## (Stretch) Deprecation of Legacy Account Creation
+
+- Once BAR on Steam is generally available, we may no longer support the
+creation of accounts through the current flow, i.e. through BYAR-Chobby via the
+[REGISTER](https://springrts.com/dl/LobbyProtocol/ProtocolDescription.html#REGISTER:client)
+command.
```

Comment:
Is this something that could impact self hosted servers? Definitely something for the very far future though.